framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -339,10 +339,13 @@ public void setRoot(CompilationUnitTree root) {
 
   @Override
   public Void scan(@Nullable Tree tree, Void p) {
-    if (tree != null && getCurrentPath() != null) {
+    if (tree == null) {
```

Comment:
What is the purpose of this change? I don't see any obvious relation to the purpose of this PR.

Reply:
There are three statements in the body.
Two of them check for `tree` being null, but the third does not.  However, it (the call to `super.scan`) also crashes when `tree` is null.  This prevents that crash.

I'm not sure when or why `tree` can be null, but this does prevent the crash.